docker/daemon/daemon_src.go

```diff
@@ -79,7 +57,7 @@ func (s *daemonImageSource) Reference() types.ImageReference {
 
 // Close removes resources associated with an initialized ImageSource, if any.
 func (s *daemonImageSource) Close() error {
-	return os.Remove(s.tarCopyPath)
+	return s.Source.Close()
```

Comment:
The `Close` method can be completely removed, Go will automatically use the method in the anonymous `Source` member.

Reply:
Should it be removed from archiveImageSource as well?

Reply:
nvm, I just saw you mentioned it in the review above.

Reply:
fixed